Test/pruning/MonomorphicSplit.bpl

```diff
@@ -0,0 +1,37 @@
+// RUN: %parallel-boogie /prune /errorTrace:0 /printPruned:"%t" "%s" > "%t"
+// RUN: %OutputCheck "%s" --file-to-check="%t-after-monomorphicSplit.bpl"
+
+// The following checks are a bit simplistic, but this is
+// on purpose to reduce brittleness. We assume there would now be two uses clauses
+// with one axiom each, and those axioms should not be conjunctions
+// (of monomorphized functions). Last one ensures that we do not have copies
+// of these axioms outside of their uses clauses.
+
+// CHECK-L: uses
+// CHECK-NEXT-L: axiom
+// CHECK-NOT-L: &&
+// CHECK-L: }
+// CHECK-L: uses
+// CHECK-NEXT-L: axiom
+// CHECK-NOT-L: &&
+// CHECK-L: }
+// CHECK-NOT: axiom
+
+// Related PR #767.
+
+function f1 <T> (x: T) : int uses {
+  axiom (forall <T> x: T :: {f1(x)} f1(x) == 42);
+}
+
+// Both f1 and the axiom will be monomorphized into bool and int instances.
+// Automatic edge inference would already ensure that only the monomorphized instances
+// are incoming, however we want to test here how the new uses clauses are determined.
+//
+// After instantiation, each instance of f1 for some T should *only* have 
+// the T-instantiated axiom as outgoing. 
+
+procedure monomorphicSplit()
+  ensures f1(true) == 42;
+	ensures f1(3) == 42;
```

Comment:
Any particular reason to indent the second ensures past the first one?

Reply:
Ah, this was not visible on my end in both Emacs and VS Code, apparently the second line was added as a tab (new laptop). Should be fixed now.